external-crates/move/move-bytecode-verifier/src/abstract_stack.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module defines an abstract stack used by various verifier passes
+
+use std::cmp::Ordering;
+use std::fmt;
+
+#[derive(Default)]
+/// An abstract value that compresses runs of the same value to reduce space usage
+pub struct AbsStack<T> {
+    values: Vec<(u64, T)>,
+}
+
+impl<T: Eq + Clone> AbsStack<T> {
+    /// Creates an empty stack
+    pub fn new() -> Self {
+        Self { values: vec![] }
+    }
+
+    /// Returns true iff the stack is empty
+    pub fn is_empty(&self) -> bool {
+        self.values.is_empty()
+    }
+
+    /// Pushes a single value on the stack
+    pub fn push(&mut self, item: T) {
+        self.push_n(item, 1)
+    }
+
+    /// Pops a single value off the stack, erroring if empty
+    pub fn pop(&mut self) -> Result<T, AbsStackError> {
+        let Some((count, last)) = self.values.last_mut() else {
+            return Err(AbsStackError::EmptyStack)
+        };
+        debug_assert!(*count > 0);
+        Ok(if *count <= 1 {
+            let (_, last) = self.values.pop().unwrap();
+            last
+        } else {
+            *count -= 1;
+            last.clone()
+        })
+    }
+
+    /// Push n copies of an item on the stack
+    pub fn push_n(&mut self, item: T, n: u64) {
+        if n == 0 {
```

Comment:
It forces a guard then on each usage and just made things look a bit more gross for cases like `VecUnpack` when the vector was empty. 
I can switch it if you feel strongly though.

Reply:
I see and good point, did not notice the function does not return an error.
I do not feel strong at all, and the "seriously?" was referred to a conceptual users doing that operation rather than my feelings on the subject which are pretty tame...